runtime/vm/monhelpers.c

```diff
@@ -150,6 +107,41 @@ objectMonitorExit(J9VMThread* vmStruct, j9object_t object)
 				}
 #endif /* JAVA_SPEC_VERSION >= 24 */
 			}
+		} else {
+			j9objectmonitor_t newLock = 0;
+			I_32 casSuccess = FALSE;
+			if (0x00 == count) {
```

Comment:
perf will be impacted since another `if` statement has been added before the flat lock monitor exit case can be reached.

Reply:
Given the switch to CAS, would the cost of an extra if statement be noticeable?